src/main/java/fr/esgi/al5_2/Tayarim/services/ProprietaireService.java

```diff
@@ -92,21 +99,22 @@ public ProprietaireDTO updateProprietaire(@NonNull Long id, @NonNull Proprietair
             throw new ProprietaireNotFoundException();
         }
 
-        if(proprietaireUpdateDTO.getEmail() != null && proprietaireRepository.findFirstByEmail(proprietaireUpdateDTO.getEmail()).isPresent()){
+        if((proprietaireUpdateDTO.getEmail() != null && !proprietaireUpdateDTO.getEmail().isBlank()) && proprietaireRepository.findFirstByEmail(proprietaireUpdateDTO.getEmail()).isPresent()){
             throw new ProprietaireInvalidUpdateBody();
         }
 
-        if(proprietaireUpdateDTO.getNumTel() != null && proprietaireRepository.findFirstByNumTel(proprietaireUpdateDTO.getNumTel()).isPresent()){
+        if((proprietaireUpdateDTO.getNumTel() != null && !proprietaireUpdateDTO.getNumTel().isBlank()) && proprietaireRepository.findFirstByNumTel(proprietaireUpdateDTO.getNumTel()).isPresent()){
             throw new ProprietaireInvalidUpdateBody();
         }
 
         Proprietaire proprietaire = optionalProprietaire.get();
 
-        proprietaire.setNom(proprietaireUpdateDTO.getNom() != null ? proprietaireUpdateDTO.getNom() : proprietaire.getNom());
-        proprietaire.setPrenom(proprietaireUpdateDTO.getPrenom() != null ? proprietaireUpdateDTO.getPrenom() : proprietaire.getPrenom());
-        proprietaire.setEmail(proprietaireUpdateDTO.getEmail() != null ? proprietaireUpdateDTO.getEmail() : proprietaire.getEmail());
-        proprietaire.setNumTel(proprietaireUpdateDTO.getNumTel() != null ? proprietaireUpdateDTO.getNumTel() : proprietaire.getNumTel());
-        proprietaire.setMotDePasse(proprietaireUpdateDTO.getMotDePasse() != null ? hashPassword(proprietaireUpdateDTO.getMotDePasse()) : proprietaire.getMotDePasse());
+        proprietaire.setNom((proprietaireUpdateDTO.getNom() != null && !proprietaireUpdateDTO.getNom().isBlank()) ? proprietaireUpdateDTO.getNom() : proprietaire.getNom());
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 172).